feat(jellyfish-api-core): enable transferdomain gov attributes

fullstackninja864 opened the pull request:
<!--  Thanks for sending a pull request! -->

#### What this PR does / why we need it:
This PR sets gov attributes to enable EVM support for playground and added dToken transfer test caes for transfer domain
#### Which issue(s) does this PR fixes?:
<!--
(Optional) Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #

#### Additional comments?:

fullstackninja864 committed:
feat(jellyfish-api-core) enable transferdomain gov attributes

fullstackninja864 committed:
fix testcase

fullstackninja864 committed:
Merge branch 'main' into harsh/set-gov-transferdomain

fullstackninja864 committed:
enabled evm-dvm

fullstackninja864 committed:
chore

fullstackninja864 committed:
updated dfi legacy address for transferdomain

lykalabrada commented on packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_account_transfer_domain.test.ts:
```diff
@@ -508,27 +521,66 @@ describe('transferDomain', () => {
     expect(tokenIdBefore0).toStrictEqual(tokenIdAfter0)
 
-    // check: dvm balance is transferred
+    // check: dev balance is updated
```
```suggestion
    // check: dvm balance is updated
```

pierregee commented on packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:
```diff
@@ -34,9 +39,18 @@ describe('TransferDomain', () => {
+    await container.call('createtoken', [{
+      symbol: 'ETH',
+      name: 'ETH',
+      isDAT: true,
```
let's also have a test for a token `isDAT: false` + loan tokens

fullstackninja864 committed:
added testcase for loand and non dtoken

fullstackninja864 committed:
updated dfid binary

fullstackninja864 committed:
Merge branch 'main' into harsh/set-gov-transferdomain

Commit:
update all test cases

Commit:
update invalid domain test cases

Commit:
update to latest stable node

Commit:
update to node rc

Commit:
remove only and add stop on container

fullstackninja864 commented on packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_account_transfer_domain.test.ts:
```diff
@@ -66,7 +96,7 @@ describe('transferDomain', () => {
   })
 
-  describe('transferDomain failed', () => {
+  describe.only('transferDomain failed', () => {
```
```suggestion
  describe('transferDomain failed', () => {
```

fullstackninja864 committed:
Update packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_account_transfer_domain.test.ts

Signed-off-by: Harsh R <53080940+fullstackninja864@users.noreply.github.com>

fullstackninja864 committed:
Merge branch 'main' into harsh/set-gov-transferdomain

Commit:
update node to beta10

chloezxyy committed:
feat(test): transfer domain for LP tokens (#2145)

<!--  Thanks for sending a pull request! -->

#### What this PR does / why we need it:

#### Which issue(s) does this PR fixes?:
<!--
(Optional) Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #

#### Additional comments?:

fullstackninja864 committed:
Merge branch 'main' of github.com:JellyfishSDK/jellyfish into harsh/set-gov-transferdomain

fullstackninja864 committed:
small fix

fullstackninja864 committed:
gov test fix

canonbrother commented on packages/jellyfish-api-core/__tests__/category/account/getTokenBalances.test.ts:
```diff
@@ -117,7 +120,7 @@ describe('Account', () => {
     await container.generate(2)
 
-    const dvmAddr = await container.call('getnewaddress', ['', 'bech32'])
+    const dvmAddr = await container.getNewAddress('dvm', 'legacy')
```
hmm.. don't get idea why change bech32 -> legacy here? 🤔

fullstackninja864 replied:
No specific reason, we have updated this line before in my transfer domain fix (only allow legacy and bech32 address ). and this line comes with conflict while rebasing, so I kept my version for consistency (As i have used same logic in transferDomain.test.ts as well)

canonbrother commented on packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:
```diff
@@ -128,7 +180,46 @@ describe('TransferDomain', () => {
+          dst: {
+            address: evmAddr, // <- not match
+            amount: '3@DFI',
+            domain: TransferDomainType.DVM // <- not match
```
need to recorrect to params here since this test is to test the dst addr type

fullstackninja864 replied:
fixed in commit: https://github.com/BirthdayResearch/jellyfishsdk/commit/a98ef131b35f43d35281f97702b491abff86bc18

canonbrother commented on packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:
```diff
@@ -128,7 +180,46 @@ describe('TransferDomain', () => {
+            domain: TransferDomainType.DVM
+          },
+          dst: {
+            address: dvmAddr, // <- not match
+            amount: '3@DFI',
+            domain: TransferDomainType.EVM // <- not match
```
recorrect params since its testing dst addr type

fullstackninja864 replied:
resolved

canonbrother commented:
one more req.. since we have this configuration
```
'v0/transferdomain/dvm-evm/src-formats': ['p2pkh', 'bech32'],
```
can you add a failed case for p2sh address in transfer domain?? 🙏

fullstackninja864 committed:
updated transferdomain testcse

canonbrother commented on packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:
```diff
@@ -128,7 +180,82 @@ describe('TransferDomain', () => {
+      await client.account.transferDomain([
+        {
+          src: {
+            address: dvmAddr, // invalid
```
```suggestion
            address: dvmAddr,
```

fullstackninja864 replied:
resolved in https://github.com/BirthdayResearch/jellyfishsdk/pull/2138/commits/abd161b451be096bc1c7018bb462b18b37fa2615

canonbrother commented on packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:
```diff
@@ -128,7 +180,82 @@ describe('TransferDomain', () => {
+            domain: TransferDomainType.EVM
+          },
+          dst: {
+            address: dvmAddr, // invalid
```
```suggestion
            address: dvmAddr,
```

canonbrother commented on packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:
```diff
@@ -128,7 +180,82 @@ describe('TransferDomain', () => {
+      await expect(promise).rejects.toThrow('Dst address must be a legacy or Bech32 address in case of "DVM" domain')
+      await container.generate(1)
+
+      // revert transfer
```
hmm.. why need revert here tho?? 🤔

fullstackninja864 replied:
There are testcase below where we are checking balance of evmAddress (considering there is no balance on evmaddress). couple of rows above we transferred 1 DFI so that we get exact error message `Dst address must be a legacy or Bech32 address in case of "DVM" domain` , to make it even I reverted the txn

fullstackninja864 replied:
resolved

canonbrother commented:
> 

mising check p2sh addr in transferdomain via txn-builder

fullstackninja864 committed:
Update packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts

Co-authored-by: canonbrother <w.canonbrother@gmail.com>
Signed-off-by: Harsh R <53080940+fullstackninja864@users.noreply.github.com>

fullstackninja864 committed:
Update packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts

Co-authored-by: canonbrother <w.canonbrother@gmail.com>
Signed-off-by: Harsh R <53080940+fullstackninja864@users.noreply.github.com>

fullstackninja864 commented:
> > 
> 
> mising check p2sh addr in transferdomain via txn-builder


here we have used p2sh address
https://github.com/BirthdayResearch/jellyfishsdk/pull/2138/files#diff-6654f415fbe9e2fcef9a31511f71ff0c857ad1d394d07abdc931d15a08d427c7R184-R185

fullstackninja864 committed:
minor fix

fullstackninja864 committed:
Merge branch 'harsh/set-gov-transferdomain' of github.com:JellyfishSDK/jellyfish into harsh/set-gov-transferdomain

fullstackninja864 committed:
removed commet

canonbrother committed:
refine and ordering
